stackslib/src/chainstate/stacks/boot/signers_voting_tests.rs

```diff
@@ -136,6 +137,420 @@ pub fn prepare_pox4_test<'a>(
+        vec![
+            (alice_principal.clone(), 1000),
+            (bob_principal.clone(), 1000),
+        ],
+        &[alice.clone(), bob.clone()],
+        Some(&observer),
+    );
+
+    // Alice and Bob will each have voted once while booting to Nakamoto
+    let alice_nonce = 1;
+    let bob_nonce = 1;
+
+    let cycle_id = current_reward_cycle;
+
+    // create vote txs
+    let alice_index = get_signer_index(&mut peer, latest_block_id, alice_address, cycle_id);
+    let bob_index = get_signer_index(&mut peer, latest_block_id, bob_address, cycle_id);
+
+    let aggregate_public_key_point: Point = Point::new();
+    let aggregate_public_key =
+        Value::buff_from(aggregate_public_key_point.compress().data.to_vec())
+            .expect("Failed to serialize aggregate public key");
+
+    let aggregate_public_key_ill_formed = Value::buff_from_byte(0x00);
+
+    let txs = vec![
+        // Alice casts a vote with a non-existant index - should return signer index mismatch error
+        make_signers_vote_for_aggregate_public_key_value(
+            alice_key,
+            alice_nonce,
+            bob_index,
+            aggregate_public_key.clone(),
+            0,
+            cycle_id + 1,
+        ),
+        // Alice casts a vote with Bobs index - should return invalid signer index error
+        make_signers_vote_for_aggregate_public_key_value(
+            alice_key,
+            alice_nonce + 1,
+            2,
+            aggregate_public_key.clone(),
+            0,
+            cycle_id + 1,
+        ),
+        // Alice casts a vote with an invalid public key - should return ill-formed public key error
+        make_signers_vote_for_aggregate_public_key_value(
+            alice_key,
+            alice_nonce + 2,
+            alice_index,
+            aggregate_public_key_ill_formed,
+            0,
+            cycle_id + 1,
+        ),
+        // Alice casts a vote with an incorrect reward cycle - should return failed to retrieve signers error
+        make_signers_vote_for_aggregate_public_key_value(
+            alice_key,
+            alice_nonce + 3,
+            alice_index,
+            aggregate_public_key.clone(),
+            0,
+            cycle_id + 2,
+        ),
+        // Alice casts vote correctly
+        make_signers_vote_for_aggregate_public_key_value(
+            alice_key,
+            alice_nonce + 4,
+            alice_index,
+            aggregate_public_key.clone(),
+            0,
+            cycle_id + 1,
+        ),
+        // Alice casts vote twice - should return duplicate vote error
+        make_signers_vote_for_aggregate_public_key_value(
+            alice_key,
+            alice_nonce + 5,
+            alice_index,
+            aggregate_public_key.clone(),
+            0,
+            cycle_id + 1,
+        ),
+        // Bob casts a vote with the wrong round - should return an invalid round error
+        make_signers_vote_for_aggregate_public_key_value(
```

Comment:
Per my above comment about round numbers, this shouldn't be a bug.  A round is simply a unique identifier for a bloc of signers.  It has no other semantic rules.

Reply:
@jferrant and I discussed this and decided to make it an error, since it should be unexpected behavior for a signer to submit a vote in a round that is more than 1 above the highest round.